torch/_inductor/select_algorithm.py

```diff
@@ -264,15 +268,57 @@ def stride(self, name, index):
             val = self.named_input_nodes[name].get_stride()[index]
         return texpr(self.rename_indexing(val))
 
-    def store_output(self, indices, val, mask):
+    def modification(self, **fixed_inputs):
+        """TODO come up with standardized way to modify the template"""
+
+        class PlaceholderSubstitution(V.WrapperHandler):  # type: ignore[name-defined]
+            self.name = "PlaceholderSubstitution"
+
+            def load(self, name: str, index: sympy.Expr):
+                if name not in fixed_inputs:
+                    # return self._inner.load(name, index)
+                    raise AssertionError(
+                        f"All loads should be coming from fixed inputs - {name}"
+                    )
+                return f"({fixed_inputs[name]})"
+
+            # TODO Doesn't work yet
+            def indirect_indexing(self, index_var, size, check):
+                return self._inner.indirect_indexing(index_var, size, False)
+                # return sympy_symbol(str(index_var))
+
+        if self.modification_cache is None:
```

Comment:
I'm not sure this is still needed. Iirc I originally added it because we went through templating twice (or something like that), but iirc @jansel changed that here https://github.com/pytorch/pytorch/pull/105987